_rules/descriptive-link-5effbb.md

```diff
@@ -0,0 +1,314 @@
+---
+id: 5effbb
+name: Link in context is descriptive
+rule_type: atomic
+description: |
+  This rule checks that all links in their context describe their purpose.
+accessibility_requirements:
+  wcag20:2.4.4: # Link Purpose (In Context)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+  - Language
+acknowledgements:
+  authors:
+    - Carlos Duarte
+    - Marie Trudelle
+    - Ramit Garg
+---
+
+## Applicability
+
+This rule applies to any HTML or SVG element which
+
+- has the [semantic role][] of ['link'][link] or a [semantic role][] that inherits from the ['link'][link] role; and
+- is [included in the accessibility tree][]; and
+- has a non-empty ("") [accessible name][].
+
+## Expectation 1
+
+The [visible text content][] of each target element together with its [programmatically determined link context][] describe the purpose of the link.
```

Comment:
This rule is making an assumption that there are two types of link texts, and that both of them have to pass the SC. There is nothing about this in the success criterion or the understanding document, so I'm not all too sure this is accurate. You also haven't addressed my previous comment about visible text content not including images.

My suggestion, drop expectation 1. I don't think there's a way to have it make sense. I get that interpreting "link text" as "the accessible name of a link" is somewhat of a stretch, but I don't think there's anything else you can do that makes more sense than that.

Reply:
I'm okay with that. With links having their accessible name calculated from content this should be a good enough approximation.

Reply:
I'm having mixed feeling about that based on techniques [ARIA7](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA7) and [ARIA8](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA8) (both used to pass 2.4.4).

---
ARIA8 has an example with description saying (emphasise mine):
> In some situations, designers may choose to lessen the visual appearance of links on a page by using shorter, repeated _link text_ such as "read more".

and passing example 
```html
<a href="taxhike.html" aria-label="Read more about Seminole tax hike">[Read more...]</a>
```

I do understand that as "link text" being the "name from content" (in that case, "[Read more…]") and the `aria-label` being part of the context.

On the other hand, this is also a sufficient technique for 2.4.9 where context is not taken into account which lead to thing that the "link text" **is** the accessible name (as suggested by @WilcoFiers). This is further supported by the fact that the WCAG def of programatically determined link content does not refer to accessible name…

😕 

---
Same phenomenon goes for ARIA7 where the example description says:
> This example will mean that the link text as shown on screen is then used as the start of the accessible name for the link.

for
```html
<h2 id="headline">Storms hit east coast</h2>

<p>Torrential rain and gale force winds have struck the east coast, causing flooding in many coastal towns.
<a id="p123" href="news.html" aria-labelledby="p123 headline">Read more...</a></p>
```

The description quite clearly separate the "link text" from the accessible name and uses the name from content as "link text"

ARIA7 further says:
> The specified behavior of aria-labelledby is that the associated label text is announced instead of the link text (not in addition to the link text). When the link text itself should be included in the label text, the ID of the link should be referenced as well in the string of IDs forming the value of the aria-labelledby attribute.

which quite strongly pushes for "link text" **not** being the accessible name.

But at the same time, the second example of ARIA7 is called "Concatenating link text from multiple sources" which hints that the "link text" is the result of the `aria-labelledby` computation.

😕 

---
I do not know how to concile the fact that WCAG both seem to call "link text" the name from content (in the sentence `shorter, repeated link text such as "read more"`) and the accessible name (in the example which passes 2.4.9 because the `aria-label` attribute is correct)…

---
Additionally, ARIA8 is sufficient for 2.4.9, meaning that `aria-label` is **not** part of the (programatically determined link) context but part of the link itself; while ARIA7 is **not** sufficient for 2.4.9, meaning that `aria-labelledby` is part of context only.

😕

Reply:
Some more thoughts on that…
I'm still not sure whether "link text" means "text content" or "accessible name" in WCAG. But I do agree that both are an accessibility issue, hence checking whether accessible name is descriptive is a valid and useful check.

I suggest to update the rule's **name** (not just description) accordingly to make it explicit that the rule only check the accessible name.

The question of whether we need or want a rule for checking whether the text content is descriptive or not is completely independant from this PR.